### PR TITLE
feat(amplify-category-auth) add support for multi env 

### PR DESCRIPTION
**Pull request by yuth**
Add auth support for multi env. Updated the cloud formation templates to create new instances of resources based on env name.

Another PR to follow adding support for env specific third party auth providers

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by haverchuck** [comment]
Approving, but there is a known issue when running amplify update on a second environment (so far reproduces when changing third party auth, but may be unrelated)

File: packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs [hunk 1]
```diff
@@ -183,9 +187,9 @@ Resources:
   # Marked as depending on UserPoolClientRole for easier to understand CFN sequencing
     Type: 'AWS::IAM::Policy'
     Properties:
-      PolicyName: !Ref userpoolClientLambdaPolicy
+      PolicyName: !If [ShouldNotCreateEnvResources, !Ref userpoolClientLambdaPolicy, !Join ['',[!Ref userpoolClientLambdaPolicy, '-', !Ref env]]]
```
Comment (kaustavghosh06) [inline packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs]: You don't need to make the policies unique. Just the role-names need to be unique.

File: packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs [hunk 2]
```diff
@@ -200,9 +204,9 @@ Resources:
   # Marked as depending on UserPoolClientLambdaPolicy for easier to understand CFN sequencing
     Type: 'AWS::IAM::Policy'
     Properties:
-      PolicyName: !Ref userpoolClientLogPolicy
+      PolicyName: !If [ShouldNotCreateEnvResources, !Ref userpoolClientLogPolicy, !Join ['',[!Ref userpoolClientLogPolicy, '-', !Ref env]]]
```
Comment (kaustavghosh06) [inline packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs]: Same as above.

File: packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs [hunk 3]
```diff
@@ -306,9 +310,9 @@ Resources:
   # Marked as depending on MFALambda for easier to understand CFN sequencing
     Type: 'AWS::IAM::Policy'
     Properties:
-      PolicyName: !Ref mfaLambdaIAMPolicy
+      PolicyName: !If [ShouldNotCreateEnvResources, !Ref mfaLambdaIAMPolicy, !Join ['',[!Ref mfaLambdaIAMPolicy, '-', !Ref env]]]
```
Comment (kaustavghosh06) [inline packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs]: Same as above.

File: packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs (hunk 1 above)
Comment (yuth) [reply packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs]: changed back

File: packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs (hunk 2 above)
Comment (yuth) [reply packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs]: changed back

File: packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs (hunk 3 above)
Comment (yuth) [reply packages/amplify-category-auth/provider-utils/awscloudformation/cloudformation-templates/auth-template.yml.ejs]: changed back

**Pull request merged**